sdk/storage/azblob/blob/models.go

```diff
@@ -194,6 +194,7 @@ type DeleteOptions struct {
 	// and all of its snapshots. only: Delete only the blob's snapshots and not the blob itself
 	DeleteSnapshots  *DeleteSnapshotsOptionType
 	AccessConditions *AccessConditions
+	BlobDeleteType   generated.DeleteType
```

Comment:
This should use the `DeleteType` from `constants.go`

Reply:
You might also want to add a comment to this field indicating the risk with permanent (like the comment you had on the API).